server/blob/blob-union/src/test/java/org/apache/james/blob/union/HybridBlobStoreTest.java

```diff
@@ -276,6 +285,58 @@ void saveInputStreamShouldRelyOnLowCostWhenSizeBasedAndBig() {
         }
     }
 
+    @Nested
+    class StoragePolicyWithHybridDuplicateWriteTests {
+
+        @Test
+        void saveShouldRelyOnPerformingAndDuplicateWriteWhenPerforming() {
+            BlobId blobId = hybridExtraStorageBlobStore.save(BucketName.DEFAULT, BLOB_CONTENT, HIGH_PERFORMANCE).block();
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(highPerformanceBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+                softly.assertThat(lowCostBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+            });
+        }
+
+        @Test
+        void saveShouldRelyOnPerformingWhenSizeBasedAndSmallAndDuplicateWrite() {
+            BlobId blobId = hybridExtraStorageBlobStore.save(BucketName.DEFAULT, BLOB_CONTENT, SIZE_BASED).block();
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(highPerformanceBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+                softly.assertThat(lowCostBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+            });
+        }
+
+        @Test
+        void saveInputStreamShouldRelyOnPerformingWhenPerforming() {
+            BlobId blobId = hybridExtraStorageBlobStore.save(BucketName.DEFAULT, new ByteArrayInputStream(BLOB_CONTENT), HIGH_PERFORMANCE).block();
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(highPerformanceBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+                softly.assertThat(lowCostBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+            });
+        }
+
+        @Test
+        void saveInputStreamShouldRelyOnPerformingWhenSizeBasedAndSmall() {
+            BlobId blobId = hybridExtraStorageBlobStore.save(BucketName.DEFAULT, new ByteArrayInputStream(BLOB_CONTENT), SIZE_BASED).block();
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(highPerformanceBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+                softly.assertThat(lowCostBlobStore.read(BucketName.DEFAULT, blobId))
+                    .hasSameContentAs(new ByteArrayInputStream(BLOB_CONTENT));
+            });
+        }
```

Comment:
You miss the test showing that upon large data & size based strategy: we should be writing only to low cost storage.